### DISCUSSION TITLE
vscode-langservers-extracted

### DISCUSSION THREAD

**Discussion by bryant-the-coder** [hrsh7th/nvim-cmp#553] [Q&A] [answered]
I think this is not a good place to ask this question but I gotta ask as it is very and super urgent. I have already run `npm i -g vscode-langservers-extracted` and input the config. Errors shows up. I am using vssnip which is great. But error shows up. I think it is a newbie problem. So help is needed

> **ERROR**
```
Error detected while processing FileType Autocommands for "html":
E5108: Error executing lua ...ppData\Local\nvim\share\nvim\runtime\lua\vim\lsp\rpc.lua:384: start `vscode-html-language-server` failed: ENOENT: no such file or directory
E5108: Error executing lua ...ppData\Local\nvim\share\nvim\runtime\lua\vim\lsp\rpc.lua:384: start `vscode-html-language-server` failed: ENOENT: no such file or directory
```

**Comment by hrsh7th** [comment]
At least, you should provide your configuration related to this problem. (Don't put your full config. Please create minimal reproduction).

What output were display with `npm ls -g | grep vscode`?

**Comment by bryant-the-coder** [comment]
`npm ls -g | grep vscode` cant work as i m on win 10. 

language_server.lua
```
--Enable (broadcasting) snippet capability for completion
local capabilities = vim.lsp.protocol.make_client_capabilities()
capabilities.textDocument.completion.completionItem.snippetSupport = true

require'lspconfig'.html.setup {}
```

cmp.lua
```
vim.g.completeopt="menu,menuone,noselect,noinsert"
local has_words_before = function()
  local line, col = unpack(vim.api.nvim_win_get_cursor(0))
  return col ~= 0 and vim.api.nvim_buf_get_lines(0, line - 1, line, true)[1]:sub(col, col):match("%s") == nil
end

local feedkey = function(key, mode)
  vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes(key, true, true, true), mode, true)
end
-- Setup nvim-cmp.
local cmp = require'cmp'
local lspkind = require('lspkind')

cmp.setup({
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },
  mapping = {
    ['<C-d>'] = cmp.mapping.scroll_docs(-4),
    ['<C-f>'] = cmp.mapping.scroll_docs(4),
    ['<C-Space>'] = cmp.mapping.complete(),
    ['<C-e>'] = cmp.mapping.close(),
    ['<CR>'] = cmp.mapping.confirm({ select = true }),
        ["<Tab>"] = cmp.mapping(function(fallback)
      if cmp.visible() then
        cmp.select_next_item()
      elseif vim.fn["vsnip#available"]() == 1 then
        feedkey("<Plug>(vsnip-expand-or-jump)", "")
      elseif has_words_before() then
        cmp.complete()
      else
        fallback()
      end
    end, { "i", "s" }),

    ["<S-Tab>"] = cmp.mapping(function()
      if cmp.visible() then
        cmp.select_prev_item()
      elseif vim.fn["vsnip#jumpable"](-1) == 1 then
        feedkey("<Plug>(vsnip-jump-prev)", "")
      end
    end, { "i", "s" }),
  },
  sources = {
    { name = 'nvim_lsp' },
    { name = 'vsnip' },
    { name = 'buffer' },
    { name = 'path'},
  },
  formatting = {
    format = lspkind.cmp_format({with_text = true, maxwidth = 50})
  }
})
```
plugins.lua
```
    -- Explorer menu
    use {
      'kyazdani42/nvim-tree.lua',
      requires = 'kyazdani42/nvim-web-devicons'
    }

    -- Treesitter
    use {
        'nvim-treesitter/nvim-treesitter',
        run = ':TSUpdate'
    }
    use {'windwp/nvim-ts-autotag'}
    use {'p00f/nvim-ts-rainbow'}
    use {'windwp/nvim-autopairs'}

    -- Statusline
    use {
      'nvim-lualine/lualine.nvim',
      requires = {'kyazdani42/nvim-web-devicons', opt = true}
    }

    -- Telescope
    use {
      'nvim-telescope/telescope.nvim',
      requires = { {'nvim-lua/plenary.nvim'} }
    }
    
    -- Bufferline
    use {'akinsho/bufferline.nvim', requires = 'kyazdani42/nvim-web-devicons'}

    -- Colorizer
    use 'norcalli/nvim-colorizer.lua'

    -- Dashboard 
    use 'glepnir/dashboard-nvim'

    -- LSP
    use 'neovim/nvim-lspconfig'
    use 'hrsh7th/nvim-cmp'
    use 'hrsh7th/cmp-nvim-lsp'
    use 'hrsh7th/cmp-buffer'
    use 'hrsh7th/cmp-path'
    use 'onsails/lspkind-nvim'
    use 'hrsh7th/cmp-vsnip'
    use 'hrsh7th/vim-vsnip'
```

**Comment by hrsh7th** [comment]
Did you check `npm ls -g`, assuming that` npm ls -g | grep vscode` doesn't work?

**Comment by bryant-the-coder** [comment]
🚀  npm ls -g
C:\Users\User\.node_modules
+-- npm@8.1.4
+-- vscode-html-languageservice@4.1.1
+-- vscode-langservers-extracted@3.0.1
`-- yarn@1.22.17

**Reply by bryant-the-coder** [reply]
Am i doing it wrong?

**Comment by bryant-the-coder** [comment]
Uhh... Hello? Or are you looking at the prob?

**Reply by hrsh7th** [reply]
Please stop requesting a reply. Has only passed only 2h yet.

**Comment by hrsh7th** [comment]
I'm improving the vscode-langservers-extracted. If this problem can't solve by this, I have no idea because I have no windows environment. Please **wait**.

If you hurry up, you can use coc.nvim.

**Reply by bryant-the-coder** [reply]
I can wait...

**Reply by hrsh7th** [reply]
I've published `v3.0.2` if this wouldn't work. I can't help you. Sorry...

**Comment by hrsh7th** [comment]
`npm uninstall -g vscode-langservers-extracted`
`npm install -g vscode-langservers-extracted@latest`

**Comment by bryant-the-coder** [comment]
So I reinstalled it. It seems like in the `bin` folder there is no script file. I don't why... But I don't know if this step is right, I git clone the repo and place it inside the `$USER\.node_modules\node_modules\vscode-langservers-extracted-master\vscode-langservers-extracted-master`. Then i will run the command in the repo as said. For your info, there is a typing error in the repo which is `ndoe ./script/depcheck.js`

**Reply by bryant-the-coder** [reply]
```
🚀  yarn install
Usage Error: The nearest package directory (C:\Users\User\.node_modules\node_modules\vscode-langservers-extracted-master\vscode-langservers-extracted-master) doesn't seem to be part of the project declared in C:\Users\User.

- If the project directory is right, it might be that you forgot to list .node_modules/node_modules/vscode-langservers-extracted-master/vscode-langservers-extracted-master as a workspace.
- If it isn't, it's likely because you have a yarn.lock or package.json file there, confusing the project root detection.

$ yarn install [--json] [--immutable] [--immutable-cache] [--check-cache] [--inline-builds] [--mode #280 
```
![image](https://user-images.githubusercontent.com/92417638/142709656-f4d44e9c-6e24-4223-807d-72af6db16790.png)
![image](https://user-images.githubusercontent.com/92417638/142709697-b66c0122-97a8-418b-9948-22894dbd3df6.png)
![image](https://user-images.githubusercontent.com/92417638/142709716-dec1081a-483f-4b10-be9a-cadca99ea123.png)

**Reply by bryant-the-coder** [reply]
They are some typo if i am not mistaken in the code

**Comment by hrsh7th** [comment]
I don't familiar with windows but I'm using the shebang in the bin file. Probably, this causes this issue.

PR welcome.

**Comment by bryant-the-coder** [comment, marked as answer]
Btw... In that repo of yours... I think for windows user... You gotta add,`.cmd` in the extracted part

**Reply by hrsh7th** [reply]
OK. I'll look into it. I don't know `cmd` file but it may be simple. (BTW, PR is really welcome :)

**Reply by bryant-the-coder** [reply]
```
require'lspconfig'.cssls.setup {
  cmd = { "vscode-css-language-server.cmd", "--stdio" },
  capabilities = capabilities,
}
```

like this the `.cmd`. i think it need to be added in windows as i watch some yt vid they use either mac or linux and none using windows... are you using linux?

**Comment by bryant-the-coder** [comment]
Created a PR in the vscode-language-server repo
